microsoft_gsl: 2017-02-13 -> 2.0.0

strager opened the pull request:
<!-- Nixpkgs has a lot of new incoming Pull Requests, but not enough people to review this constant stream. Even if you aren't a committer, we would appreciate reviews of other PRs, especially simple ones like package updates. Just testing the relevant package/service and leaving a comment saying what you tested, how you tested it and whether it worked would be great. List of open PRs: <https://github.com/NixOS/nixpkgs/pulls>, for more about reviewing contributions: <https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#sec-reviewing-contributions>. Reviewing isn't mandatory, but it would help out a lot and reduce the average time-to-merge for all of us. Thanks a lot if you do! -->
###### Motivation for this change

GSL now has an official release. Upgrade microsoft_gsl to the latest release (made August 20, 2018).

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

strager committed:
microsoft_gsl: 2017-02-13 -> 2.0.0

GSL now has an official release. Upgrade microsoft_gsl to the latest
release (made August 20, 2018).

joachifm commented:
@GrahamcOfBorg build microsoft_gsl

joachifm commented on pkgs/development/libraries/microsoft_gsl/default.nix:
```diff
@@ -1,23 +1,23 @@
-  version = "2017-02-13";
+  version = "2.0.0";
 
   src = fetchgit {
```
Please use the source archive instead of `fetchgit` (from a cursory glance it does not appear to me that `fetchgit` will do so by itself).

strager replied:
Good catch!

strager committed:
microsoft_gsl: use fetchFromGitHub

fetchFromGitHub has friendlier ways to download sources. Use it instead
of fetchgit for microsoft_gsl.

c0bw3b commented:
@GrahamcOfBorg build microsoft_gsl
